CMakeLists_files.cmake

```diff
@@ -181,6 +181,7 @@ list (APPEND MAIN_SOURCE_FILES
   opm/simulators/utils/ComponentName.cpp
   opm/simulators/utils/DeferredLogger.cpp
   opm/simulators/utils/FullySupportedFlowKeywords.cpp
+  opm/simulators/utils/InstantiationMarcros.hpp
```

Comment:
Macros. Also I think the name is a bit too generic, and/or the macros themself is too generic. maybe INSTANTIATE_TYPE_INDICES or the likes.

Reply:
Thanks for the comments, it is indeed too generic and automatic name changing changed a lot of macro usage. I had to roll back to do it manually. 

I updated it now and please have a look.